lib/job-iteration/iteration.rb

```diff
@@ -176,7 +176,13 @@ def assert_implements_methods!
         )
       end
 
-      unless respond_to?(:build_enumerator, true)
+      if respond_to?(:build_enumerator, true)
+        parameters = method(:build_enumerator).parameters
```

Comment:
This is not strictly related to your PR or needs addressing, but it's somewhat pointless to check for arguments and the method signature on _every_ `perform` of a job. Once a job class has been validated, we know it implements methods right. This makes me believe that we should be skipping `assert_implements_methods!` for classes that has already been validated.

Reply:
That is a great point, we could have a local cache of checked classes and avoid having to call `assert_implements_methods!` unnecessary. I will explore this on another PR.